todoapp/app/src/test/java/com/example/android/architecture/blueprints/todoapp/data/source/TasksRepositoryTest.java

```diff
@@ -233,35 +234,35 @@ public void activateTaskId_activatesTaskToServiceAPIUpdatesCache() {
     public void getTask_requestsSingleTaskFromLocalDataSource() {
         // Given a stub completed task with title and description in the local repository
         Task task = new Task(TASK_TITLE, "Some Task Description", true);
-        setTaskAvailable(mTasksLocalDataSource, task);
+        Optional<Task> taskOptional = Optional.of(task);
+        setTaskAvailable(mTasksLocalDataSource, taskOptional);
```

Comment:
how about constructing the optional in the `setTaskAvailable` and have the following method signature:
`setTaskAvailable(TasksDataSource dataSource, Task taskOptional)`

Reply:
I see your point, but I am using the constructed optional in the assertion below.